libtelemetry/src/main/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -488,7 +490,12 @@ private PermissionCheckRunnable obtainPermissionCheckRunnable() {
   }
 
   private void startLocation() {
-    applicationContext.startService(obtainLocationServiceIntent());
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      applicationContext.startForegroundService(obtainLocationServiceIntent());
```

Comment:
You could additionally ensure that the service is not started from the background by wrapping this line with `ProcessLifecycleOwner.get().lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)`.

Reply:
Nice catch.

Reply:
starting a long- lived foreground service should only be done when absolutely necessary. It will trigger the OS (Android 8) to show a notification "Application X" is consuming your battery.

Reply:
@florianPOLARSTEPS that is correct, but I am killing the service when the app enters the background to avoid this issue. When returning to the foreground, the service is spun up again and continues collecting and processing event data.

Reply:
@electrostat Alright, thanks... I just noticed the LifecyleObserver, sorry about that.